File: StoryFlow_Shared/Models/GetUserDto.cs
using StoryFlow_Shared.Enums;

namespace StoryFlow_Shared.Models
{
    public class GetUserDto
    {
        public int Id { get; set; }
        public string? Email { get; set; }
        public string? FirstName { get; set; }
        public string? Nick { get; set; }
        public Gender? Gender { get; set; }
        public DateOnly? DateOfBirth { get; set; }
        public int? Stars { get; set; }
        public int Tickets { get; set; }
        public ICollection<GetHobbyDto> Hobbies { get; set; } = new List<GetHobbyDto>();
        public GetUserPreferencesDto? UserPreferences { get; set; }
    }
}

File: StoryFlow_Client/src/app/models/get-user-dto.ts
import { GetUserPreferencesDto } from "./get-user-preferences-dto";

export interface GetUserDto {
    id: number;
    email: string | null;
    firstName: string | null;
    nick: string | null;
    gender: number | null;
    dateOfBirth: string | null;
    stars: number | null;
    tickets: number;
    hobbies: [];
    userPreferences: GetUserPreferencesDto;
}